src/service/storage_arango.py

```diff
@@ -839,3 +858,67 @@ async def import_bulk_ignore_collisions(self, arango_collection: str, documents:
+        """
+        Save an internal selection to the database.
+        """
+        await self._insert_model(
+            selection, selection.internal_selection_id, COLL_SRV_INTERNAL_SELECTIONS)
+
+    async def get_selection_internal(self, internal_selection_id: str) -> models.InternalSelection:
+        """ Get an internal selection. """
+        col = self._db.collection(COLL_SRV_INTERNAL_SELECTIONS)
+        doc = await col.get(internal_selection_id)
+        if doc is None:
+            raise errors.NoSuchSelectionError(
+                f"There is no internal selection {internal_selection_id}")
+        doc[models.FIELD_SELECTION_STATE] = models.ProcessState(doc[models.FIELD_SELECTION_STATE])
+        return models.InternalSelection.construct(
+            **models.remove_non_model_fields(doc, models.InternalSelection))
+
+    async def save_selection_active(self, selection: models.ActiveSelection):
+        """
+        Save an active selection to the database.
+        """
+        # might want to return the old one so the active internal ID can be logged?
+        await self._insert_model(
+            selection, selection.selection_id_hash, COLL_SRV_ACTIVE_SELECTIONS)
+
+    async def get_selection_active(self, selection_id: str) -> models.ActiveSelection:
+        """
+        Get an active selection given a selection_id. Note that selection IDs are session tokens,
+        and so are omitted from error messages.
+        """
+        # tried to DRY this up with get internal selection but got too messy
+        col = self._db.collection(COLL_SRV_ACTIVE_SELECTIONS)
+        doc = await col.get(selection_id)
+        if doc is None:
+            raise errors.NoSuchSelectionError(
+                f"There is no selection by the given selection ID.")
+        return models.ActiveSelection.construct(
+            **models.remove_non_model_fields(doc, models.ActiveSelection))
+
+
+    async def update_selection_active_last_acess(self, selection_id_hash, last_access):
+        """
+        Update the last time the active selection was accessed.
+        Throws an error if the selection doesn't exist.
+
+        selection_id_hash - the hashed ID / token for the selection.
+        last_access - the time at which the selection was accessed in epoch milliseconds.
+        """
+        await self._update_last_access(
+            selection_id_hash,
+            COLL_SRV_ACTIVE_SELECTIONS,
+            last_access,
+            errors.NoSuchSelectionError,
+            # selection IDs are tokens, don't stick them in exception messages, dummy
+            errstr="No selection found for the provided selection ID"
+        )
+
+    async def _insert_model(self, model: BaseModel, key: str, collection: str, overwrite=False):
+        doc = jsonable_encoder(model)
+        doc[FLD_ARANGO_KEY] = key
+        col = self._db.collection(collection)
+        await col.insert(doc, overwrite=overwrite, silent=True)
```

Comment:
reminds me `silent` means insert errors are ignored? is that desired?

Reply:
silent just means nothing is returned: https://aioarango.readthedocs.io/en/latest/specs.html#aioarango.collection.StandardCollection.insert